### PR TITLE
fix: make commands.(bot_)has_permissions work without the bot in server

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

If the channel is not cached (or the bot in not in the server) this will make has_permissions work on interactions.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
This is blocked by a potential API bug.

**Comment by onerandomusername** [comment]
Created an issue for the API discrepancy: https://github.com/discord/discord-api-docs/issues/5316

**Comment by onerandomusername** [comment]
API discrepancy has been resolved.

File: disnake/ext/commands/core.py [hunk 1]
```diff
@@ -2129,7 +2129,10 @@ async def test(ctx):
 
     def predicate(ctx: AnyContext) -> bool:
         ch = ctx.channel
-        permissions = ch.permissions_for(ctx.author, ignore_timeout=False)  # type: ignore
+        if isinstance(ctx, disnake.Interaction):
+            permissions = ctx.permissions
+        else:
+            permissions = ch.permissions_for(ctx.author, ignore_timeout=False)  # type: ignore
```
Comment (shiftinv) [inline disnake/ext/commands/core.py:2135]:
```suggestion
        if isinstance(ctx, disnake.Interaction):
            permissions = ctx.permissions
        else:
            ch = ctx.channel
            permissions = ch.permissions_for(ctx.author, ignore_timeout=False)  # type: ignore
```

File: disnake/ext/commands/core.py [hunk 2]
```diff
@@ -2218,7 +2221,10 @@ def bot_has_permissions(**perms: bool) -> Callable[[T], T]:
         raise TypeError(f"Invalid permission(s): {', '.join(invalid)}")
 
     def predicate(ctx: AnyContext) -> bool:
-        permissions = ctx.channel.permissions_for(ctx.me, ignore_timeout=False)  # type: ignore
+        if isinstance(ctx, disnake.Interaction):
+            permissions = ctx.app_permissions
+        else:
+            permissions = ctx.channel.permissions_for(ctx.me, ignore_timeout=False)  # type: ignore
```
Comment (shiftinv) [inline disnake/ext/commands/core.py:2227]:
```suggestion
        if isinstance(ctx, disnake.Interaction):
            permissions = ctx.app_permissions
        else:
            ch = ctx.channel
            permissions = ch.permissions_for(ctx.me, ignore_timeout=False)  # type: ignore
```

File: changelog/673.bugfix.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+|commands| Change :func:`has_permissions <ext.commands.has_permissions>` and :func:`bot_has_permissions <ext.commands.bot_has_permissions>` checks to work in guilds that only added the ``applications.commands`` scope.
```
Comment (shiftinv) [inline changelog/673.bugfix.rst:1]:
```suggestion
|commands| Change :func:`has_permissions <ext.commands.has_permissions>` and :func:`bot_has_permissions <ext.commands.bot_has_permissions>` checks to work with interations in guilds that only added the ``applications.commands`` scope, and in DMs.
```

File: disnake/ext/commands/core.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/ext/commands/core.py]: https://github.com/DisnakeDev/disnake/pull/673/commits/ef48e17b11447d59c03768ef51af9d7d6c9e78bf

File: disnake/ext/commands/core.py (hunk 2 above)
Comment (onerandomusername) [reply disnake/ext/commands/core.py]: https://github.com/DisnakeDev/disnake/pull/673/commits/ef48e17b11447d59c03768ef51af9d7d6c9e78bf

**Review by shiftinv — APPROVED** [review]
looks good to me, thanks

**Pull request merged**